webapp/src/Controller/API/ContestController.php

```diff
@@ -263,6 +263,113 @@ public function setBannerAction(Request $request, string $cid, ValidatorInterfac
         return new Response('', Response::HTTP_NO_CONTENT);
     }
 
+    /**
+     * Delete the text for the given contest.
+     */
+    #[IsGranted('ROLE_ADMIN')]
+    #[Rest\Delete('/{cid}/text', name: 'delete_contest_text')]
+    #[OA\Response(response: 204, description: 'Deleting text succeeded')]
+    #[OA\Parameter(ref: '#/components/parameters/cid')]
+    public function deleteTextAction(Request $request, string $cid): Response
+    {
+        $contest = $this->getContestAndCheckIfLocked($request, $cid);
+        $contest->setClearContestText(true);
+        $contest->processContestText();
+        $this->em->flush();
+
+        $this->eventLogService->log('contests', $contest->getCid(), EventLogService::ACTION_UPDATE,
+            $contest->getCid());
+
+        return new Response('', Response::HTTP_NO_CONTENT);
+    }
+
+    /**
+     * Set the text for the given contest.
+     */
+    #[IsGranted('ROLE_ADMIN')]
+    #[Rest\Post("/{cid}/text", name: 'post_contest_text')]
+    #[Rest\Put("/{cid}/text", name: 'put_contest_text')]
+    #[OA\RequestBody(
+        required: true,
+        content: new OA\MediaType(
+            mediaType: 'multipart/form-data',
+            schema: new OA\Schema(
+                required: ['text'],
+                properties: [
+                    new OA\Property(
+                        property: 'text',
+                        description: 'The text to use, as either text/html, text/plain or application/pdf.',
```

Comment:
```suggestion
                        description: 'The text to use, as either text/html, text/plain or application/pdf. Can be used to upload all problem statements in one file for participants to use.',
```

Reply:
I don't like this wording, since it makes you think we split the PDF, which we don't

Reply:
@nickygerritsen my problem is that `test` is very generic, `problem-text` is something about problems which is reasonable clear that its not that public, but `contest-text` IMO is for a first time person reading the API maybe:
- A text about the contest?
- A nice welcome text banner?
- General hints?
- Rules for the contest?

So although I'm fine with not accepting my text, I would have liked it if we could be more clear, but as its already merged I'm not sure what we should do.

Reply:
I agree with @vmcj that `contest-text` is not a good name. For example `problemset-text` would be better. That makes it clear that this is the complete problemset and not split out in separate problem texts.

Reply:
There is a new PR that addresses this, please check it. We came up with problemset as well.

Reply:
Ok, I see this is already being addressed in #2402